### ISSUE TITLE
Modify layout to add custom key/value pairs

### ISSUE THREAD

**Issue by markus1189** [majikthys/log4j2-logstash-jsonevent-layout#30]
Currently I am using `Markers` with Logback to add key/value pairs directly to the logged json as described in the logstash-logback-encoder docu here: https://github.com/logstash/logstash-logback-encoder#loggingevent_custom_event.
The json output from the logback logger is later used with logstash, so it has to be compatible with their layout.

However, because of <reasons> we have to switch from Logback to Log4j2.

After a little bit of research it seems like using `ThreadContext.put` (https://logging.apache.org/log4j/2.x/manual/thread-context.html) is the normal way to add such key value pairs to logging events.  Unfortunately, I cannot figure out how to include them in the generated jsonevent-layout when using this library.

After configuring `log4j2-logstash-jsonevent-layout` the output looks like this:

```
{
  "@version" : "1",
  "@timestamp" : "2016-04-13T17:15:21.903+02:00",
  "timeMillis" : 1460560521903,
  "thread" : "ForkJoinPool-3-worker-7",
  "level" : "WARN",
  "loggerName" : "...",
  "message" : "...",
  "endOfBatch" : false,
  "loggerFqcn" : "org.apache.logging.slf4j.Log4jLogger"
}
```

but I would like to add keys and values to get output like this:

```
{
  "@version" : "1",
  "@timestamp" : "2016-04-13T17:15:21.903+02:00",
  "timeMillis" : 1460560521903,
  "thread" : "ForkJoinPool-3-worker-7",
  "level" : "WARN",
  "loggerName" : "...",
  "message" : "...",
  "endOfBatch" : false,
  "loggerFqcn" : "org.apache.logging.slf4j.Log4jLogger",
  "user_key1": "user_value1",
  ...
  "user_keyx": "user_valuex"
}
```

Trying to add them via `ThreadContext.put("user_key1")` leads to 

```
2016-04-13 16:39:10,209 application-akka.actor.default-dispatcher-14 ERROR An exception occurred processing Appender console java.lang.UnsupportedOperationException
        at java.util.Collections$UnmodifiableMap.putAll(Collections.java:1463)
        at org.apache.logging.log4j.core.layout.LogStashJSONLayout.toSerializable(LogStashJSONLayout.java:175)
        at org.apache.logging.log4j.core.layout.LogStashJSONLayout.toSerializable(LogStashJSONLayout.java:41)
        at org.apache.logging.log4j.core.layout.AbstractStringLayout.toByteArray(AbstractStringLayout.java:148)
        at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.append(AbstractOutputStreamAppender.java:112)
        at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:152)
        at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:125)
        at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:116)
        at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
        at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:390)
        at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:378)
        at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:362)
        at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:352)
        ... and more
```

is this #27?

TL;DR: The questions:

(1) is `ThreadContext` the right thing to achieve this?
(2) how can I configure the json layout to include keys/values (if 1) then the ones added via `ThreadContext.put`)

**Comment by majikthys** [comment]
Not supported yet, but we can add fairly easily. Feel free to submit a pull request
